state/model.go

```diff
@@ -228,20 +228,21 @@ func (st *State) ModelExists(uuid string) (bool, error) {
 	return count > 0, nil
 }
 
-// ModelActive returns true if a model with the supplied UUID exists
-// and is not being imported as part of a migration.
-func (st *State) ModelActive(uuid string) (bool, error) {
+// ModelActive returns true and the model type if a model
```

Comment:
Please don't overload methods. The model type has nothing to do with whether it's active or not. Create a new method.

Reply:
Ok, but we need to know those 2 bits of info at the same time. So we're adding an extra mongo read which introduces load for little benefit. I could rename the method

Reply:
In the *current* code base, we have one place where we call ModelActive, and will also need to know the model type. Lumping together unrelated information in this method is going to cause more harm than an extra mongo read. Calling it "ModelActiveAndModelType" is not really any better.

An alternative would be to pass the StatePool into this worker, get a *state.Model, and use its methods to get the bits that are all read in at once.